modules/misc/xdg-autostart.nix

```
@@ -35,6 +51,9 @@ in {
   };
 
   config = mkIf (cfg.enable && cfg.entries != [ ]) {
-    xdg.configFile = listToAttrs (map mapDesktopEntry cfg.entries);
+    xdg.configFile = if cfg.readOnly then {
+      autostart.source = linkedDesktopEntries;
+    } else
+      listToAttrs (map mapDesktopEntry cfg.entries);
```

Using the `recurisive` option for HM's file options would be cleaner, I think. Recursive = true would link all files inside `source` recursively, whereas recursive = false would link `autostart` directly to `source`.

This also allows us to remove `mapDesktopEntry` above.

```suggestion
    xdg.configFile."autostart" = {
      recursive = !cfg.readOnly;
      source = linkedDesktopEntries;
    };
```

---

Thanks for the feedback, using `recursive` didn't occur to me but that does seem to work as intended so I've changed the implementation and added a test.